\documentclass[12pt]{article}
\usepackage[margin = 2.4cm]{geometry} % For margins of 3cm
\usepackage{graphicx}
\usepackage{float} % For H float position
\usepackage{gensymb} % For some symbols
\usepackage{amsfonts, amssymb, amsmath} % All three for maths symbols
\usepackage[export]{adjustbox} % For figure frames
\setlength{\parskip}{6pt} % To make nice looking paragraph spacing
\usepackage[export]{adjustbox} % For figure frames
\usepackage{rotating}
\usepackage[section]{placeins}
\usepackage{setspace} % For double spacing
\usepackage{pdfpages} % Allows including PDFs
\usepackage[sort&compress]{natbib} % bibliographies
\doublespacing

\title{Spatiotempotal properties of NDD genes during brain development for biomarker discovery}
\author{Urwah Nawaz}


\begin{document}

\begin{titlepage}
    \maketitle
\end{titlepage}

\paragraph{What would the reader gain from this lit review}

~\\ The reader will have understanding of the following: 

\begin{itemize}
	\item Introduction to brain development 
	\item Introduction to neurodevelopmental disorders
	\item Introduction to sequencing technologies for gene expression studies 
	\item an understanding of how gene expression studies contribute to our knowledge of brain development and NDDs
	\item Available resources
\end{itemize}


\section{Aims of the project}

\begin{itemize}
	\item To comprehensively characterize the expression properties of ID and CP enes in the normal human brain
	\begin{itemize}
		\item To determine whether ID and CP genes are expressed in a cell-type speciic mannner using single-cell RNA-seq data
		\item To characterize the developmental trajectory of gene expression for ID and CP genes and assess their expression during cellular maturation in brain organiods and assess their expression across multiple fetail developmental periods 
		\item  To characterize the spatial and temporal properties of ID and CP gene expression in the adult brain, by assessing age-dependent changes  brain-region and cortical layer-specificity 
		
	\end{itemize}
	
	\item To determine whether convergent gene expression changes are observed in ID and CP in patient-derived cells with heterogenous mutations 
	
\end{itemize}

\section{Introduction}
Human brain development is a complex and a tightly regulated process during which changes occur at both anatomical and functional levels.
The processes of brain development are highly dependent on the appropriate expression of RNA and proteins, 
Mutations that result in altered expression or function of these gene products can cause or contribute to neurodevelopmental disorders (NDDs). 



\section{Neurodevelopmental disorders}

Neurodevelopmental disorders (NDDs) are a group of early onset neurological disorders affecting an estimated 10\% to 15\% of the population. Common NDDs include autism spectrum disorders, intellectual disability, epilepsy, motor/tic movement disorders among others. 



Common NDDs include autism spectrum disroder, intellectual disability, epilepsy and motor/tic movement disorders, and are characterized by strong clinical co-morbidity which suggests common genetic etiology. 

\begin{itemize}
	\item ASD and how many genes have been identified 
	\item ID and how many genes have been identified 
	\item Other NDDs and how many genes have been identified 
\end{itemize}
The genetic heterogeneity and overlap observed in NDDs make it difficult to identify the genetic causes of specific clinical symptoms

\paragraph{Autism}

~\\ Autism spectrum disorders represent genetically heterogeneous group of neurodevelopmental syndromes with high prevalence that has a wide range of phenotype. While there is no unifying hypothesis about the molecular pathology of autism, it is clear that the disorder is highly hertiable and results from the combination of genetic, neurologic , immunologic and environmental factors. 

Recent advnces in sequencing technologies have made it possible to gain insight into the molecular aspects of ASD. Microarray technologies and next-generation sequencing have enabled high-throughput discovery of genes likely to be involved in the molecular pathology of autism.5, 6, 7, 8 However, as the success in discovery has risen, the number of candidate genes with associated risk for ASD has also stretched well into the hundreds.9, 10 As of December 2014, 667 genes have been implicated in autism. Despite the large amounts of data now available, the general lack of replication across studies suggests that more data will be needed to fully characterize the genetic models responsible for the various forms of autism.

\subsection{Heterogeneity of NDDs}


\section{Expression studies}

\subsection{RNA: bulk vs single-cell}
The human brain contains billions of highly differentiated and interconnected cells that form intricate neural networks and collectively control the physical activities and high-level cognitive functions, such as memory, decision-making, and social behavior. Big data is required to decipher the complexity of cell types, as well as connectivity and functions of the brain. The newly developed single-cell sequencing technology, which provides a comprehensive landscape of brain cell type diversity by profiling the transcriptome, genome, and/or epigenome of individual cells, has contributed substantially to revealing the complexity and dynamics of the brain and providing new insights into brain development and brain-related disorders.

The development of single-cell technologies, especially single-cell RNA-sequencing (scRNA-seq), has provided new opportunity to address this challenge by looking through transcriptomic profile of each individual cell. Since the first introduction of scRNA-seq technique by Tang et al. in 2009. In recent years, a dozen of scRNA-seq studies that look into the cellular composition, heterogeneity, and disease-specific populations in mammalian brain has also demonstrated the power of this technology in addressing the challenges in understanding the complexity, connectivity, and functions of brain cell types 

\subsection{Co-expression network analysis}

\section{Characterisation of brain diversity using single-cell}
\subsection{Studies that have characterized the single-cell transcriptome in the brain}

\begin{table}[]
	\begin{tabular}{llllllll}
		Year & Cells reported  & Method  & Technique  & Species  &  Cell isolation & Brain region & Developmental stages\\
		2014 &  &  &  & & & & \\
		&  &  &  & &&& \\
		&  &  &  & &&&
	\end{tabular}
\end{table}


what to review

\begin{itemize}
	\item human brain single cell rna-seq studies 
	\item reread papers used for PhD proposal 
	\item number and types of samples 
	\item method used for sequencing 
	\item data availablity 
	\item main conclusions of the studies
	\item co-expression networks and their utility in spatiotemporal studes 
	\item this kind of analysis in other areas (particularly cancer)
	\item Use of single-cell analysis vs bulk rna-seq 
\end{itemize}


\begin{itemize}
	
	\item what are the datasets available 
	\item where am i going to get the genelists from 
	\item what gene ontologies am I getting with my genes 
	\item current co-expression networks that are available
	\item lack of ID gene networks 
	\item benchmark potentially?
	\item Issues with bulk rna? 
	
\end{itemize}






\end{document}
