---
title: "Neurodevelopmental gene expression signatures"
author: "Urwah Nawaz"
date: "15 January 2020"
output:
  html_document:
    code_folding: hide
    toc: true
    toc_float: true 
    theme: lumen
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval = TRUE, cache = TRUE, cache.lazy = FALSE)
```

# Paper summary
The study by Velmeshev et al (2019) analysed transcriptomic changes in the brains of individuals with autism to gain insight into the molecular pathology of ASD in a cell-type specifc manner.

The study used single-nucleus RNA-sequencing strategy on postmortem tissue samples including the prefrontal cortex and anerior cingulator cortex from 15 ASD patients and 16 controls. None of the ASD patients were diagnosed with ID. Additionally, patients that only had sporadic epilespy were also included in this analysis along with 7 age matched controls.

## Characteristics of the data 

* Generated 104,559 single-nuclei gene expression profiles 

   * 52,556 from control subjects 
   * 52,003 from ASD patients 

* Detected a median of 1391 genes and 2213 transcripts per nucleus
* Identified 17 cell types by using expression of known cell markers
* Neurons expressed more genes and transcripts than glia 

## Differential expression results 

The study aimed to identify differentially expressed genes in cell-type specific manner by comparing nuclear profiles from ASD and controls using a linear mixed model. 

* Detected 692 differential expression events (q value < 0.05; expression level change > 10%) in 513 unique DEGs
* 76% of these were differentisally expressed in a single cell type
* Top differentially expressed genes in non-neuronal cell types were up-regulated in protoplasmic astrocytes and microglia 
* 75 DEGs were found in the SFARI database
* SFARI genes were most overrepresented in L2/3 and L4 excitatory neurons, followed by VIP and somatostatin expression interneurons 

Gene expression was also investigated in neuronal and glial subtypes. Over all this analysis revealed dysregulated development and synaptic signaling in components of upper-layer cortical circuitry as well as changes in the cellular state of microglia and protoplasmic astrocytes 

# Aim of analysis

To use a single-nucleus data from ASD cases and controls and calculate a "neurodevelopmental gene dysregulation score" for each ASD case and each control. 

Each of the genes may only show a minor change and different genes may be changed in different patients, however when we look at them as a group, we might see a cumulative effect. 


# Method Aims 

* Download data from Vekmeshev et al and create a expression matrix 

* Make a list of neurodevelopmental disease genes with subcategories

* Load data into R and perform an exploratory analysis 

* For each NDD gene 
    + Calculate the z-score of its expression in a given individual and cell type relative to all cells from controls in the same cell-type 
    + Box-plot these z-scores 
    + Calculate a wilcoxon test comparing NDD genes and non-NDD genes for each individual  


# Analysis 

## Gene lists

For this analysis, we want to investigate the expression of  NDD genes in ASD individuals compared to controls. 

Neurodevelopmental disorder gene lists: 

```{r gene list setup}
suppressPackageStartupMessages({
  library(pander)
  library(kableExtra)
  library(tibble)
  library(magrittr)
  library(dplyr)
  library(rmarkdown)
  library(gdata)
  library(biomaRt)
  library(EnsDb.Hsapiens.v79)
})
```

* **ASD related genes**

List of genes was downloaded from SFARI database and includes 913 genes in total

```{r}
ASD_genes <- read.csv("~/Documents/Lists/SFARI-Gene_genes_01-03-2020release_02-06-2020export.csv")


ASD_genes %>%
  dplyr::select("gene.symbol", "gene.name", "ensembl.id", 
                "chromosome", "gene.score", "number.of.reports") %>%
  kable(escape = F, align = "c") %>%
  kable_styling(c("striped", "condensed"), full_width = F) %>%
  kable_styling(bootstrap_options = "striped", full_width = F) %>%
  scroll_box(width = "1000px", height = "200px")
  

```

The ASD risk genes from SFARI are scored from 1 to 3, where 1 is high confidence genes and 3 means that there is suggestive evidence for ASD risk.





* **Intellectual disability genes**

The ID gene list was download from UMC 

```{r ID gene}
ID_genes <- read.xls("~/Documents/Lists/NDD_genes.xlsx", sheet = "UMC ID", header= FALSE) %>% 
  as.data.frame() %>% 
  dplyr::select(V1)


geneSymbols <- as.character(ID_genes$V1)
geneIDs <- ensembldb::select(EnsDb.Hsapiens.v79, keys= geneSymbols, keytype = "SYMBOL", columns = c("SYMBOL","GENEID"))
ID_genes <- merge(as.data.frame(ID_genes), 
                  as.data.frame(geneIDs), by.x = "V1",
                  by.y = "SYMBOL", sort=FALSE)


ID_genes %<>% 
  dplyr::select("Gene" = "V1", "GENEID") %>% dplyr::filter(str_detect(GENEID, "ENSG"))
```

   * **Epilpsy genes**
   
   The Epilepsy genes have been curated in house after an in-depth literature review. 
```{r}
Epi_genes <- read.xls("~/Documents/Lists/Final_Epil_list_annotated_Jan19.xls")  %>% 
  as.data.frame() 

geneSymbols <- as.character(Epi_genes$old.list)
geneIDs <- ensembldb::select(EnsDb.Hsapiens.v79, keys= geneSymbols, keytype = "SYMBOL", columns = c("SYMBOL","GENEID"))

Epi_genes_complete <- merge(as.data.frame(Epi_genes), 
                           as.data.frame(geneIDs), by.x = "old.list",
                           by.y = "SYMBOL", sort=FALSE)


Epi_genes_complete %>% 
  as.data.frame() %>%
  dplyr::select("Gene" = "old.list", "GENEID", "Type") %>%
  kable(escape = F, align = "c") %>%
 kable_styling(c("striped", "condensed"), full_width = F) %>% kable_styling(bootstrap_options = "striped", full_width = F) %>%
  scroll_box(width = "1000px", height = "200px")
  
```



There are overlapping genes for each category, however for the purpose of this analysis, we will keep the groups as they are.

## Exploratory analysis

 In this section, we explore several properties of the sn-RNAseq data including mean expression and variation of expression of various genes. Raw data from UCSC cell browser was downloaded and loaded into R. A SingleCellExperiment object was created and the data was filtered and normalized.

```{r libraries}
suppressPackageStartupMessages({
  library(stringr) # CRAN
  library(Biobase) # Bioconductor 
  library(SingleCellExperiment) # bioC
  library(scater) # bioC
  library(readr) # CRAN
  library(ggplot2) # Bioc
  library(ggpubr) # bioC
  library(Matrix) # CRAN
  library(DropletUtils) #bioC
  library(tidyverse)
  library(gdata)
  library(EnsDb.Hsapiens.v79)
  library(scran)
  library(tibble)
  library(data.table)
})
```
 

```{r meta}
meta <- read.table("~/Documents/Data/Autism/meta.tsv", header=TRUE, sep = "\t", as.is = TRUE, row.names = 1)
```

Number of cells belonging to each cluster:

```{r, fig.height=6, fig.width=6}
meta %>%
  group_by(cluster) %>%
  summarise(counts = n()) %>% 
  ggplot(aes(x = cluster, y = counts)) +
  geom_bar(fill = "#0073C2FF", stat = "identity") +
  geom_text(aes(label = counts), vjust = -0.3) + coord_flip() +
  theme_pubclean()
```


### Quality control and explatory analysis of the data

```{r loading data}
sn_counts <- readMM("~/Documents/Data/Autism/matrix.mtx")
genes <- read_tsv("~/Documents/Data/Autism/genes.tsv", col_names = FALSE)
barcodes <- read_tsv("~/Documents/Data/Autism/barcodes.tsv", col_names=FALSE)
```

```{r sce object}
sce <- SingleCellExperiment(
  assays=list(counts = sn_counts),
  rowData = genes$X1 ,
  colData = barcodes,
  metadata = meta
)

rownames(sce) <- genes$X1

```


Here, we will use the `perCellQCMetrics` to calculate metrics for each cell in the data. 
This gives us an output of the following:

* Sum: the total number of counts in a cell (i.e library size)

* Detected: number of features for the cell that have counts above the detection limit (default was used: 0)

* subsets_x_percentage: percentage of all counts that come from the control sets 


In this case, it would be interesting to see the proportion of NDD genes in the dataset with respect to detection and sum. 

```{r per Cell QC}
per.cell <- perCellQCMetrics(sce, 
                             subsets=list(ASD=ASD_genes$ensembl.id, 
                                          EP_genes=factor(Epi_genes$GENEID), 
                                          ID_genes=factor(ID_genes_complete$GENEID)))

colData(sce) <- cbind(colData(sce), per.cell)
```


```{r histograms, fig.cap="Histograms of library sizes (left) and the number of expressed genes for all cells in the snRNA-seq data"}

par(mfrow=c(1,2))
hist(sce$sum/1e6,xlab="Library sizes (millions)", 
     main="",breaks=20, col="grey80", ylab="Number of cells")
hist(sce$detected, xlab="Number of features detected", 
     main="",breaks=20, col="grey80", ylab="Number of cells", ylim = c(0,40000), 
     xlim = c(0, 12000))
```

#### Proportion of NDD genes detected in this analysis 

```{r}
par(mfrow=c(1,2))
hist(sce$subsets_ASD_percent, xlab="Proportion of ASD genes (%)", 
     main="", col="pink1", ylab="Number of cells")
hist(sce$subsets_ID_genes_percent,  xlab="Proportion of ID genes (%)", 
     main="", col="grey80", ylab="Number of cells")
```

```{r}
hist(sce$subsets_EP_genes_percent) ### not sure why this isnt working 
```

### Per feature QC

* mean: the mean counts for each feature

* detected: the percentage of detection of features 


```{r, cache = TRUE}
per.feat <- perFeatureQCMetrics(sce,subsets=list(ASD=ASD_genes$ensembl.id,
                                                 EP_genes=factor(Epi_genes$GENEID),
                                                 ID_genes=factor(ID_genes_complete$GENEID)))

rowData(sce) <- cbind(rowData(sce), per.feat)
```

Here, the scatterplot is the mean plotted aganist the detected for all cells. 

```{r}
plotRowData(sce, y="detected", x="mean") + scale_x_log10()
```



```{r}
sce <- logNormCounts(sce)
cpm(sce) <- calculateCPM(sce)
```




# Notes 

## Single-nucleus RNA-seq 

Single-nucleus RNA-seq strategies involve analyzing the nuclei instead of cells. This method is advantegeous when attempting to profile expression in cells that are difficult to isolate such as frozen tissue samples. 

More reading here:

https://www.biocompare.com/Editorial-Articles/557506-Advantages-of-Single-Nucleus-RNA-Sequencing/

https://jasn.asnjournals.org/content/30/1/23



## Normalization in sc-RNAseq 

Various normalization methods exist for bulk RNA-seq which can be applied to sc-RNAseq, however sources of variation specific to sc-RNAseq such as technical dropouts make it necessary to have normalization methods specific to sc-RNAseq 

* Most commonly used normalization method includes CPM normalization 

* CPM normalization assumes that all cells in the dataset initially contain an equal number of mRNA molecules and count depth differences arise only due to sampling. 

* After normalization, data matrices are typically log(x+1) transformation. This transformation has three main effects:

   * Distances between log-transformed expression values represent log fold changes 
   * Log transformations mitigates mean variance relationship in sc data
   * Log transformation reduces the skewness of the data to approximate the assumption of a data that is normally distributed